### PR TITLE
Recalculate base price when ticket price changes | Closes #2807

### LINKED ISSUE #2807: Ticket price change does not reflect in TPC
Two issues:
- editing ticket total price from ticket card or ticket edit form does not reflect when calculator is opened.
- editing ticket total price in card or table does not recalculate the base price.

Task:
- Recalculate base price when the price of a ticket is changed in card/modal
- Set `reverseCalculate` to true

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This fixes the bug to update related base price when ticket price changes:
- Creates `useRecalculateBasePrice` hook
- Wires ticket price update to recalculate base price

### CODE REVIEW

File: assets/src/domain/eventEditor/ui/tickets/hooks/useRecalculateBasePrice.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		const newPrices = calculateBasePrice(getDataState(null));
+		// the price if present should be the basePrice
+		const [basePrice] = newPrices.filter(isBasePrice);
+
+		// if we are lucky
+		if (basePrice?.id) {
+            const { id, amount } = basePrice;
+            // update the base price
+			updateEntity({ id, amount });
+		}
```
Comment (tn3rb) [inline assets/src/domain/eventEditor/ui/tickets/hooks/useRecalculateBasePrice.ts:25]: super nitpicky comment, but the formatting looks off on this... has a mix of spaces and tabs and they don't appear to line up

File: assets/src/domain/eventEditor/ui/tickets/hooks/useEditTicketModal.ts [hunk 2]
```diff
@@ -19,7 +24,13 @@ const useEditTicketModal: FormModal = ({ entityId, entityDbId }) => {
 
 	const onSubmit = useCallback<FormProps<UpdateTicketInput>['onSubmit']>(
 		(fields): void => {
-			updateEntity(fields);
+			const priceHasChanged = fields.price !== ticket?.price;
+			// if price has changed, reverseCalculate must be enabled
+			const reverseCalculate = priceHasChanged ? true : fields?.reverseCalculate;
+			updateEntity({ ...fields, reverseCalculate });
+			if (priceHasChanged) {
```
Comment (tn3rb) [inline assets/src/domain/eventEditor/ui/tickets/hooks/useEditTicketModal.ts:31]: 👍

File: assets/src/domain/eventEditor/ui/tickets/hooks/useRecalculateBasePrice.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { usePriceMutator } from '@edtrServices/apollo/mutations';
+
+const useRecalculateBasePrice = (ticketId: EntityId): VoidFunction => {
+	// This will give us the exact state expected by `calculateBasePrice()`
+	const getDataState = useInitialState({ ticketId });
+	const { updateEntity } = usePriceMutator();
+
+	return useCallback(() => {
+		// get the list of updated prices with the amount of base price updated
+		const newPrices = calculateBasePrice(getDataState(null));
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/tickets/hooks/useRecalculateBasePrice.ts:16]: `useInitialState` callback receives `initialState` and then it got destructured, it's a bit confusing
image

Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/tickets/hooks/useRecalculateBasePrice.ts:16]: `null` and `undefined` values are no destructured.

Comment (nicolad) [reply assets/src/domain/eventEditor/ui/tickets/hooks/useRecalculateBasePrice.ts:16]: makes sense, that null just caught my attention and I was thinking if it's a good idea to just call `getDataState` w/o any args and then do the check inside of `useInitialState`, smth like:
```
return useCallback<StateInitializer>(
		(initialState) => {
			return { ...(initialState && { ...initialState }), ticket, prices };
		},
		[ticket, prices]
	);
```
but I don't like very much this approach :point_up: 
so, just wrote my thoughts outloud

File: assets/src/domain/eventEditor/ui/tickets/hooks/useEditTicketModal.ts (hunk 2 above)
Comment (nicolad) [reply assets/src/domain/eventEditor/ui/tickets/hooks/useEditTicketModal.ts:31]: agree, this reads so easy

Comment (nicolad) [reply assets/src/domain/eventEditor/ui/tickets/hooks/useEditTicketModal.ts:31]: `if (fields.price !== ticket?.price)` - would make no sense to me

**Pull request merged**